frontend/packages/console-app/src/components/nodes/NodeTerminal.tsx

```diff
@@ -0,0 +1,165 @@
+import * as React from 'react';
+import { Alert } from '@patternfly/react-core';
+import { Firehose, FirehoseResult, LoadingBox } from '@console/internal/components/utils';
+import { NodeKind, PodKind } from '@console/internal/module/k8s';
+import { PodExecLoader } from '../../../../../public/components/pod';
+import { ImageStreamTagModel, PodModel } from '../../../../../public/models';
+import { k8sCreate, k8sGet, k8sKillByName } from '../../../../../public/module/k8s';
+
+type NodeTerminalErrorProps = {
+  error: string;
+};
+
+type NodeTerminalInnerProps = {
+  obj?: FirehoseResult<PodKind>;
+};
+
+type NodeTerminalProps = {
+  obj: NodeKind;
+};
+
+// TODO: use https://github.com/openshift/console/blob/a5bc8fce01761043023a187e50baef1374b9013e/frontend/packages/dev-console/src/utils/shared-submit-utils.ts#L149-L154 instead
```

Comment:
Planning to do this as a follow-on as there are 8 files unrelated to this PR that will need to be updated.